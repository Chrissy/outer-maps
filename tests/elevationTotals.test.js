import React from "react";
import ElevationTotals from "../components/elevationTotals";
import renderer from "react-test-renderer";

const elevations = [
  {
    coordinates: [-121.01849819661, 48.5847955640306],
    id: 141044,
    index: 0,
    elevation: 985,
    elevationGain: 0,
    elevationLoss: 0,
    distanceFromPreviousPoint: 0
  },
  {
    coordinates: [-121.018136436568, 48.5846454373507],
    id: 141044,
    index: 1,
    elevation: 987,
    elevationGain: 2,
    elevationLoss: 0,
    distanceFromPreviousPoint: 31.42249865416899
  },
  {
    coordinates: [-121.017988374201, 48.5845484843434],
    id: 141044,
    index: 2,
    elevation: 989,
    elevationGain: 2,
    elevationLoss: 0,
    distanceFromPreviousPoint: 15.32921380080007
  }
];

it("renders", () => {
  const tree = renderer
    .create(<ElevationTotals elevations={elevations} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
