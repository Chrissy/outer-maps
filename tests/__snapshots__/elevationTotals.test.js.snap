// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
.emotion-55 {
  margin-bottom: 0.5em;
}

.emotion-38 {
  display: grid;
  grid-template-columns: 30% 0.35fr 0.35fr 0.29fr;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media (max-width:950px) {
  .emotion-38 {
    grid-template-columns: 0.29fr 0.26fr 0.26fr 0.19fr;
    font-size: 0.875em;
  }
}

@media (max-width:350px) {
  .emotion-38 {
    font-size: 0.75em;
  }
}

.emotion-11 {
  width: 100%;
  height: 100%;
  width: 140%;
  height: 140%;
  position: relative;
  z-index: 1;
  -webkit-transform: translate(-7%,2%);
  -ms-transform: translate(-7%,2%);
  transform: translate(-7%,2%);
}

@media (max-width:750px) {
  .emotion-11 {
    width: 124%;
    height: 124%;
    -webkit-transform: translate(-5%,-2%);
    -ms-transform: translate(-5%,-2%);
    transform: translate(-5%,-2%);
  }
}

.emotion-0 {
  stroke: #D8D8D8;
  stroke: #3232FF;
  stroke-width: 5;
  fill: #fff;
}

.emotion-2 {
  stroke: #D8D8D8;
  stroke-width: 5;
  fill: none;
}

.emotion-4 {
  font-size: 7pt;
  fill: #838383;
}

.emotion-6 {
  font-size: 32pt;
  fill: #3232FF;
  font-weight: 700;
  line-height: 0;
}

.emotion-8 {
  fill: #3232FF;
  font-weight: 600;
  font-size: 10pt;
  line-height: 0;
}

.emotion-20 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
  border-color: #D8D8D8;
  border-width: 0;
  margin: 1em 0 0.75em;
  border-style: solid;
  height: 4em;
}

.emotion-15 {
  font-size: 1.5em;
  font-weight: 700;
  dislay: flex;
  -webkit-align-items: top;
  -webkit-box-align: top;
  -ms-flex-align: top;
  align-items: top;
}

.emotion-13 {
  font-size: 0.33em;
}

.emotion-17 {
  font-size: 0.875em;
  color: #838383;
  font-size: 0.66em;
  line-height: 1.4;
  color: inherit;
}

.emotion-29 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
  border-color: #D8D8D8;
  border-width: 0 1px;
  margin: 1em 0 0.75em;
  border-style: solid;
  height: 4em;
}

.emotion-53 {
  margin-top: 0.25em;
  position: relative;
  width: 100%;
  line-height: 0;
  padding: 0 1em;
  box-sizing: border-box;
  width: 100%;
}

.emotion-42 {
  width: 100%;
}

.emotion-40 {
  fill: #FF5500;
  mix-blend-mode: multiply;
}

.emotion-50 {
  position: absolute;
  top: 0;
  left: 0;
  width: calc(100% - 2em);
  height: 100%;
  margin: 0 1em;
}

.emotion-44 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: #5800EC;
  padding: 0 1.25em;
  font-size: 0.5em;
  font-weight: 700;
  position: absolute;
  left: 0%;
  top: 75%;
  -webkit-transform: translate(-30%,35%);
  -ms-transform: translate(-30%,35%);
  transform: translate(-30%,35%);
  border-radius: 1.5em;
  min-width: 25px;
  color: #fff;
  height: 1.75em;
  border: 2px solid #fff;
}

.emotion-44:after {
  display: block;
  content: "";
  width: 0;
  height: 0;
  border-left: 5px solid transparent;
  border-right: 5px solid transparent;
  border-top: 5px solid #5800EC;
  position: absolute;
  left: 30%;
  top: 100%;
  -webkit-transform: translate(-50%,-10%);
  -ms-transform: translate(-50%,-10%);
  transform: translate(-50%,-10%);
}

.emotion-46 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: #5800EC;
  padding: 0 1.25em;
  font-size: 0.5em;
  font-weight: 700;
  position: absolute;
  left: 67.21143890597577%;
  top: 37.5%;
  -webkit-transform: translate(-50%,35%);
  -ms-transform: translate(-50%,35%);
  transform: translate(-50%,35%);
  border-radius: 1.5em;
  min-width: 25px;
  color: #fff;
  height: 1.75em;
  border: 2px solid #fff;
}

.emotion-46:after {
  display: block;
  content: "";
  width: 0;
  height: 0;
  border-left: 5px solid transparent;
  border-right: 5px solid transparent;
  border-top: 5px solid #5800EC;
  position: absolute;
  left: 50%;
  top: 100%;
  -webkit-transform: translate(-50%,-10%);
  -ms-transform: translate(-50%,-10%);
  transform: translate(-50%,-10%);
}

.emotion-48 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: #5800EC;
  padding: 0 1.25em;
  font-size: 0.5em;
  font-weight: 700;
  position: absolute;
  left: 100%;
  top: 0%;
  -webkit-transform: translate(-70%,35%);
  -ms-transform: translate(-70%,35%);
  transform: translate(-70%,35%);
  border-radius: 1.5em;
  min-width: 25px;
  color: #fff;
  height: 1.75em;
  border: 2px solid #fff;
}

.emotion-48:after {
  display: block;
  content: "";
  width: 0;
  height: 0;
  border-left: 5px solid transparent;
  border-right: 5px solid transparent;
  border-top: 5px solid #5800EC;
  position: absolute;
  left: 70%;
  top: 100%;
  -webkit-transform: translate(-50%,-10%);
  -ms-transform: translate(-50%,-10%);
  transform: translate(-50%,-10%);
}

<div
  className="emotion-55 emotion-56"
>
  <div
    className="emotion-38 emotion-39"
  >
    <svg
      className="emotion-10 emotion-11 emotion-12"
      viewBox="0 0 105 105"
    >
      <circle
        className="emotion-0 emotion-1"
        cx={52.5}
        cy={52.5}
        r={50}
      />
      <path
        className="emotion-2 emotion-3"
        d="M 102.5 52.5 A 50 50 0 0 1 102.40133642141357 55.63952597646568"
        transform="rotate(-90 52.5 52.5)"
      />
      <text
        alignmentBaseline="middle"
        className="emotion-4 emotion-5"
        textAnchor="middle"
        x="50%"
        y="25%"
      >
        Difficulty
      </text>
      <text
        alignmentBaseline="middle"
        className="emotion-6 emotion-7"
        textAnchor="middle"
        x="50%"
        y="52%"
      >
        0
      </text>
      <text
        alignmentBaseline="middle"
        className="emotion-8 emotion-9"
        textAnchor="middle"
        x="50%"
        y="74%"
      >
        Very Easy
      </text>
    </svg>
    <div
      className="emotion-19 emotion-20 emotion-21"
    >
      <div
        className="emotion-15 emotion-16"
      >
        +13
        <sup
          className="emotion-13 emotion-14"
        >
          ft
        </sup>
      </div>
      <div
        className="emotion-17 emotion-18"
      >
        Elevation Gain
      </div>
    </div>
    <div
      className="emotion-19 emotion-29 emotion-21"
    >
      <div
        className="emotion-15 emotion-16"
      >
        -0
        <sup
          className="emotion-13 emotion-14"
        >
          ft
        </sup>
      </div>
      <div
        className="emotion-17 emotion-18"
      >
        Elevation Loss
      </div>
    </div>
    <div
      className="emotion-19 emotion-20 emotion-21"
    >
      <div
        className="emotion-15 emotion-16"
      >
        0
      </div>
      <div
        className="emotion-17 emotion-18"
      >
        Miles
      </div>
    </div>
  </div>
  <div
    className="emotion-52 emotion-53 emotion-54"
  >
    <svg
      className="emotion-42 emotion-43"
      viewBox="0 0 275 110"
    >
      <g />
      <polyline
        className="emotion-40 emotion-41"
        points="0,110 0,107.5 184.83145699143336,66.25 275,25 275,110"
      />
    </svg>
    <div
      className="emotion-50 emotion-51"
    >
      <div
        className="emotion-44 emotion-45"
        x={0}
        y={75}
      >
        985
      </div>
      <div
        className="emotion-46 emotion-45"
        x={67.21143890597577}
        y={37.5}
      >
        987
      </div>
      <div
        className="emotion-48 emotion-45"
        x={100}
        y={0}
      >
        989
      </div>
    </div>
  </div>
</div>
`;
